IdentityCore/src/util/NSDictionary+MSIDExtensions.m

```diff
@@ -239,4 +239,25 @@ - (id)msidObjectForKey:(NSString *)key ofClass:(Class)requiredClass
     return nil;
 }
 
+- (NSMutableDictionary *) mutableDeepCopy
+{
+    NSMutableDictionary *returnDict = [[NSMutableDictionary alloc] initWithCapacity:[self count]];
+    NSArray *keys = [self allKeys];
+    for (id key in keys)
+    {
+        id value = [self valueForKey:key];
+        id copy = nil;
+        if ([value respondsToSelector:@selector(mutableDeepCopy)])
+            copy = [value mutableDeepCopy];
+        else if ([value respondsToSelector:@selector(mutableCopy)])
+            copy = [value mutableCopy];
+        if (copy == nil)
+            copy = [value copy];
+        
+        [returnDict setObject:copy forKey:key];
```

Comment:
better to do nil check before we call `setObject`? :)

Reply:
I think this should be fine as the copy should return a non nil object.